tsql/TSqlParser.g4

```diff
@@ -194,6 +202,187 @@ another_statement
     | transaction_statement
     | use_statement
     ;
+//https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-application-role-transact-sql
+
+alter_application_role
+    : ALTER APPLICATION ROLE application_role=id  alter_application_role_with_clause
+    ;
+
+alter_application_role_with_clause
+    :WITH
+       (NAME EQUAL new_application_role_name=id COMMA?)
+       (PASSWORD EQUAL  application_role_password=STRING COMMA?)?
+       (DEFAULT_SCHEMA  EQUAL app_role_default_schema=id COMMA?)?
+    ;
+
+alter_assembly
+    : ALTER ASSEMBLY assembly_name=id alter_assembly_clause 
+    ;
+
+alter_assembly_clause
+    : alter_assembly_from_clause alter_assembly_with_clause alter_assembly_drop_clause alter_assembly_add_clause
+    ;
+
+alter_assembly_from_clause
+    : FROM (client_assembly_specifier | assembly_bits )
+    |
```

Comment:
Please use optional `?` operator in super root rule instead of empty alternative. I mean
```
alter_assembly_clause
    : alter_assembly_from_clause? alter_assembly_with_clause alter_assembly_drop_clause alter_assembly_add_clause
    ;

alter_assembly_from_clause
    : FROM (client_assembly_specifier | assembly_bits )
    ;
```